### PR TITLE
SONAR-6258 Persist highlighting into file sources

### PR DESCRIPTION
**Pull request by julienlancelot**
SONAR-6258 Persist highlighting into file sources

### CODE REVIEW

**Comment by GithubIntegration** [comment]
SonarQube analysis reported 15 new issues:
* 2 critical
* 10 major
* 3 info

**Comment by GithubIntegration** [comment]
SonarQube analysis reported 5 new issues:
* 2 major
* 3 info

File: server/sonar-server/src/test/java/org/sonar/server/computation/step/PersistFileSourcesStepTest.java [hunk 1]
```diff
@@ -157,34 +157,19 @@ public void persist_last_line() throws Exception {
...
-    FileUtils.writeLines(sourceFile, Lists.newArrayList("line\t1", "line2"));
+    initBasicReport(2);
 
     sut.execute(new ComputationContext(new BatchReportReader(reportDir), ComponentTesting.newProjectDto(PROJECT_UUID)));
 
     assertThat(dbTester.countRowsOfTable("file_sources")).isEqualTo(1);
     FileSourceDto fileSourceDto = dbClient.fileSourceDao().select("FILE");
     assertThat(fileSourceDto.getLineHashes()).isEqualTo("137f72c3708c6bd0de00a0e5a69c699b\ne6251bcf1a7dc3ba5e7933e325bbe605\n");
-    assertThat(fileSourceDto.getSrcHash()).isEqualTo("fe1ac2747e8393015698f2724d8d2835");
+    assertThat(fileSourceDto.getSrcHash()).isEqualTo("4fcc82a88ee38e0aa16c17f512c685c9");
```
Comment (teryk) [inline server/sonar-server/src/test/java/org/sonar/server/computation/step/PersistFileSourcesStepTest.java]: out of curiosity, why the change of hash ?

**Comment by teryk** [comment]
lgtm-no-comment-awesome

File: server/sonar-server/src/test/java/org/sonar/server/computation/step/PersistFileSourcesStepTest.java (hunk 1 above)
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/computation/step/PersistFileSourcesStepTest.java]: Because the content of lines from the report has changed (see initBasicReport()) -> it's now line1 (without the tab character), line2

**Comment by GithubIntegration** [comment]
SonarQube analysis reported 18 new issues:
* 1 blocking
* 11 major
* 1 minor
* 5 info
